src/odemis/driver/orsay.py

```diff
@@ -2648,30 +2681,324 @@ class Scanner(model.Emitter):
     to update the pixel size.
+
     """
 
     def __init__(self, name, role, parent, **kwargs):
         """
         Defines the following VA's and links them to the callbacks from the Orsay server:
         • power: IntEnumerated, choices={0: "off", 1: "on"}
         • blanker: VAEnumerated, choices={True: "blanking", False: "no blanking", None: "imaging"}
+        • horizontalFov: FloatContinuous, unit="m", range=(0.0, 1.0)
+        • scale: VAEnumerate, choices={(1.0, 1.0), (2.0, 2.0)}
+        • resolution: TupleContinuous Int, unit="px", range=[(1, 1), (1024, 1024)]
+        • translation: TupleContinuous Float, unit="px", range=[(-512.0, -512.0), (512.0, 512.0)]
+        • rotation: FloatContinuous, unit="rad", range=(-pi, pi)
+        • acceleratorVoltage: FloatContinuous, unit="V", range=(0.0, 3e4)
+        • dwellTime: FloatEnumerated, unit="s", choices=(1e-3, 5e-4, 1e-4, 5e-5, 1e-5, 5e-6, 1e-6, 5e-7, 2e-7, 1e-7)
+        • pixelSize: TupleContinuous, unit="m", range=((0.0, 0.0), (1 / 1024, 1 / 1024)))
         """
-
         super().__init__(name, role, parent=parent, **kwargs)
+        self._shape = (1024, 1024)
 
+        if not hasattr(self.parent, "_fib_beam"):
+            raise ValueError(f"To create a Orsay scanner component the parent should also have the FIBBeam child.")
         self._fib_beam = self.parent._fib_beam  # reference to the FIBBeam object
+
+        if not hasattr(self.parent, "_fib_source"):
+            raise ValueError(f"To create a Orsay scanner component the parent should also have the FIBSource child.")
         self._fib_source = self.parent._fib_source  # reference to the FIBSource object
-        # In case there is no such sibling, let it raise an exception
 
         self.blanker = self._fib_beam.blanker
         self.power = self._fib_source.gunOn
 
+        self.horizontalFov = self._fib_beam.horizontalFov
+        self.scale = model.VAEnumerated((self.shape[0] / self._fib_beam.imageFormat.value[0],
+                                         self.shape[1] / self._fib_beam.imageFormat.value[1]),
+                                        unit="", choices={(1.0, 1.0), (2.0, 2.0)},
+                                        setter=self._setScale)
+        self.scale.subscribe(self._listenerImageFormat)  # Subscribe to update the scale when the image format changes.
+
+        self.resolution = self._fib_beam.resolution
+        self.translation = self._fib_beam.translation
+
+        self.rotation = self._fib_beam.rotation
+        self.accelVoltage = self._fib_source.acceleratorVoltage
+
+        self.dwellTime = self._fib_beam.dwellTime
+        self.pixelSize = model.TupleContinuous((self._fib_beam.horizontalFov.value / self.shape[0],
+                                                self._fib_beam.horizontalFov.value / self.shape[1]),
+                                               unit="m", range=((0.0, 0.0), (1 / 1024, 1 / 1024)))
```

Comment:
There is more love needed here. The rules for the pixel size are the following:
* .pixelSize should be read-only. (and if you want, you can even skip the range, and use just a VigilantAttribute). It contains the horizontalFoV / shape. It should be updated whenever horizontalFoV changes.
* MD_PIXEL_SIZE should be set on the metadata, and contain a similar value *but* "scale * horizontalFoV / shape". It should be updated whenever horizontalFoV or scale change.